Lesson17/baljan/contactBook.js

```diff
@@ -0,0 +1,201 @@
+/*
+===========================================================
+  SIMPLE CONTACT BOOK APPLICATION
+===========================================================
+In this project, you'll create a Contact Book to store and
+manage basic info: name, phone, and email.
+
+You'll practice:
+1. Arrays and objects
+2. Loops (for, for-of, findIndex, etc.)
+3. Conditionals (if-else)
+4. Basic CRUD (Create, Read, Update, Delete) functionality
+
+Below is a step-by-step guide, with sample code and
+comments explaining what each section does. Run this file
+in Node.js or in your browser's console to see the output.
+*/
+
+/*
+-----------------------------------------------------------
+  STEP 1: Setup and Initial Data
+-----------------------------------------------------------
+1. Create an array named 'contacts' with a few initial
+   sample contacts.
+2. Each contact is an object with 'name', 'phone', and
+   'email' properties.
+*/
+
+const contacts = [
+  { name: "Dilek", phone: "0 224 555 88 77", email: "xyz@yahoo.com" },
+  { name: "Ali", phone: "0 225 555 88 77", email: "xds@yahoo.com" },
+  { name: "Ayşe", phone: "0 224 555 88 55", email: "sdf@yahoo.com" },
+];
+
+/*
+-----------------------------------------------------------
+  STEP 2: Display All Contacts
+-----------------------------------------------------------
+Function: displayAllContacts()
+- Loops over the 'contacts' array.
+- Logs a descriptive string for each contact.
+
+Example output:
+  Name: Alice, Phone: 123-456-7890, Email: alice@example.com
+*/
+
+function displayAllContacts(contactList, uppercaseMode) {
+  for (const contact of contactList) {
+    if (uppercaseMode) {
+      console.log(
+        `NAME: ${contact.name}, PHONE:${contact.phone}, EMAIL:${contact.email}`,
+      );
+      continue;
+    } else {
+      console.log(
+        `name: ${contact.name}, phone:${contact.phone}, email:${contact.email}`,
+      );
+    }
+  }
+}
+
+/*
+-----------------------------------------------------------
+  STEP 3: Add a New Contact
+-----------------------------------------------------------
+Function: addContact(name, phone, email)
+- Creates a new contact object and pushes it into 'contacts'.
+- Checks if a contact with the same name already
+  exists before adding. If found, logs a warning and returns.
+- Logs "Contact added successfully." if everything is good.
+*/
+
+function addContact(name, phone, email, contactList) {
+     if (findContact(name, contactList))  {
```

Comment:
You updated the code correction so it uses the `findContact` function. However, the `findContact` function itself is not defined in the file. Make sure that you copy it from the original file.